demos/TicTacToe/Classes/MainMenuScene.h

```diff
@@ -1,19 +1,92 @@
+// Copyright 2020 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 #ifndef TICTACTOE_DEMO_CLASSES_MAINMENU_SCENE_H_
 #define TICTACTOE_DEMO_CLASSES_MAINMENU_SCENE_H_
 
 #include "cocos2d.h"
+#include "firebase/auth.h"
+#include "firebase/database.h"
+#include "firebase/future.h"
+#include "firebase/util.h"
 #include "ui/CocosGUI.h"
+using std::to_string;
+
+void LogMessage(const char*, ...);
+void ProcessEvents(int);
+void WaitForCompletion(const firebase::FutureBase&, const char*);
+std::string GenerateUid(std::size_t);
```

Comment:
If these functions are only called in MainMenuScene.cpp then you you should declare them in MainMenuScene.cpp and not here. This file is mainly for those other parts of code that will use MainMenuScene and need to know how to use it, internal methods should be kept out of the public space as much as possible.  For instance, what if I had another source file that also wanted to define a funciton LogMessage?

Reply:
Just to recap from our chat, I will be moving these into a util.h and util.cpp shared file seeing multiple programs use these files.